logger_test.go

```diff
@@ -0,0 +1,36 @@
+package analytics
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"testing"
+)
+
+// This test ensures that the interface doesn't get changed and stays compatible
```

Comment:
I think this is the point, we want it to break at compile time to ensure that the interface stays compatible. For example if we introduce a method in the `Logger` interface that doesn't exist in `*testing.T` then we want something to tell us we're probably getting ourselves in trouble.